Core/src/Material/AverageMaterials.cpp

```diff
@@ -29,41 +29,42 @@ Acts::MaterialSlab Acts::detail::combineSlabs(const MaterialSlab& slab1,
   // to properly account for the energy loss in multiple material.
 
   // use double for (intermediate) computations to avoid precision loss
+  const auto thickness1 = static_cast<double>(slab1.thickness());
+  const auto thickness2 = static_cast<double>(slab2.thickness());
```

Comment:
I guess sonar recommended this? I'm not a huge fan of `auto` for very simple types like this, but ok.

Reply:
Yes, sonar. I found the explicit type also clearer. I would be happy to revert it ♡⸜(˶˃ ᵕ ˂˶)⸝♡

Reply:
I reverted it, the basic type seems much more readable.